### PR TITLE
Refactor transformations

### PR DESCRIPTION
**Pull request by rasmusgo**
* Rename transforms to `LocalTransform` and `GlobalTransform`
* Rename related systems.
* ~~Add `update_global_transforms_system` that calls `update_transform_matrix_system` and 
`update_parent_transform_matrix_system`. This is because they don't make much sense to run separately.~~

### CODE REVIEW

**Comment by rasmusgo** [comment]
I'm also used to "world". I took the names that @kanerogers suggested. I would happily change to `WorldTransform` or even `WorldFromLocal`.

File: hotham/src/components/local_transform.rs [hunk 1]
```diff
@@ -40,8 +40,8 @@ impl LocalTransform {
         }
     }
 
-    /// Convenience function to convert the `Transform` into a `nalgebra::Isometry3`
-    pub fn position(&self) -> Isometry3<f32> {
+    /// Convenience function to convert the `Transform` into a `nalgebra::Isometry3` without applying scale.
+    pub fn rigid_part(&self) -> Isometry3<f32> {
```
[the remark this answers is not in the document]
Comment (rasmusgo) [reply hotham/src/components/local_transform.rs:44]: This is to emphasize that we are omitting the scale here. The "rigid" in rigid body and in rigid transformation has the same meaning (=rotation and translation only), so the association is good.

**Comment by rasmusgo** [comment]
I think it could make sense to keep the names `GlobalTransform` and `LocalTransform` for the components while still using the terms world space and local space for the members, eg:

```rs
pub struct GlobalTransform {
    pub world_from_local: Matrix4<f32>,
}
```

This would also open up for storing decompositions or timestamps/flags to avoid recomputing stuff.

File: hotham/src/components/local_transform.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (kanerogers) [reply hotham/src/components/local_transform.rs:44]: I'm with @sinoth on this one `position` is better here, `rigid` immediately (though incorrectly) makes me think of a `RigidBody`

**Comment by kanerogers** [comment]
First of all, fabulous work as always @rasmusgo and thanks for reviewing this one @sinoth.

I'll exercise my BDFL powers here and.. agree with you both. I think `World` is better than `Global`, in hindsight, and **yes, everything must be 100% consistent**

**Comment by kanerogers** [comment]
In a stunning about face, I have now flipped a coin and decided on `Global`!

**Review by kanerogers — APPROVED** [review]
do it

**Pull request merged**